x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/alert_renderer/helpers/index.test.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+
+import type { Ecs } from '../../../../../../../../common/ecs';
+import { TestProviders } from '../../../../../../../common/mock';
+import { TimelineId } from '../../../../../../../../common/types';
+import { AlertFieldFlexGroup, DEFAULT_GAP, eventKindMatches, showWith } from '.';
+
+describe('helpers', () => {
+  describe('eventKindMatches', () => {
+    test('it returns true when `eventKind` is an array of (just) `signal`', () => {
+      const eventKind = ['signal'];
+
+      expect(eventKindMatches(eventKind)).toBe(true);
+    });
+
+    test('it returns true when `eventKind` is an array of (just) mixed-case `sIgNaL`', () => {
+      const eventKind = ['sIgNaL'];
+
+      expect(eventKindMatches(eventKind)).toBe(true);
+    });
+
+    test('it returns true when `eventKind` is an array containing other values, AND a mixed-case `sIgNaL`', () => {
+      const eventKind = ['foo', 'bar', 'sIgNaL', '@baz'];
```

Comment:
These tests go to great lengths 😂 . On a more serious question, with the whole transition to `alerts` and `kibana.alert`, do you happen to be aware if there are plans to move away from `signal`

Reply:
The _Elastic Common Schema reference_ [entry for `event.kind`](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-kind) defines the set of valid values for `event.kind`:

>**Important**: The field value must be one of the following:
>
>alert, enrichment, event, metric, state, pipeline_error, signal

I'm not aware of any plans to switch the value of `event.kind` for  Detection Engine alerts from `signal` to `alert`, because `alert` is defined in https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-kind.html as the value for _external_ alerts:

> ## alert
>
> This value indicates an event such as an alert or notable event, triggered by a detection rule executing externally to the Elastic Stack.
>
> `event.kind:alert` is often populated for events coming from firewalls, intrusion detection systems, endpoint detection and response systems, and so on.

`signal` is defined [here](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-kind.html#ecs-event-kind-signal) as the value used by Elastic solutions:

> ## signal
>
>This value is used by Elastic solutions (e.g., Security, Observability) for alert documents that are created by rules executing within the Kibana alerting framework.
>
>Usage of this value is reserved, and data ingestion pipelines must not populate `event.kind` with the value "signal".